sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1548,6 +1548,29 @@ object PushPredicateThroughNonJoin extends Rule[LogicalPlan] with PredicateHelpe
         filter
       }
 
+    case filter @ Filter(condition, w: Window) if w.partitionSpec.isEmpty =>
+      w.windowExpressions match {
+        case Seq(alias @ Alias(WindowExpression(_: RowNumber, WindowSpecDefinition(Nil, orderSpec,
+            SpecifiedWindowFrame(RowFrame, UnboundedPreceding, CurrentRow))), _)) =>
+          val aliasAttr = alias.toAttribute
+          val limitValue = splitConjunctivePredicates(condition).collectFirst {
+            case LessThanOrEqual(e, IntegerLiteral(v)) if e.semanticEquals(aliasAttr) => v
+            case EqualTo(e, IntegerLiteral(v)) if e.semanticEquals(aliasAttr) => v
+            case LessThan(e, IntegerLiteral(v)) if e.semanticEquals(aliasAttr) => v - 1
```

Comment:
`GreaterThan` and `GreaterThanOrEqual` would also work with the literal on the left